packages/api/src/controllers/access-control.ts

```diff
@@ -50,73 +57,182 @@ app.post(
+        if (!req.body.pub) {
+          console.log(`
+            access-control: gate: no pub provided for playbackId ${playbackId}, disallowing playback
+          `);
+          throw new ForbiddenError(
+            "Content is gated and requires a public key"
+          );
+        }
+
+        const query = [];
+        query.push(sql`signing_key.data->>'publicKey' = ${req.body.pub}`);
+        const [signingKeyOutput] = await db.signingKey.find(query, {
+          limit: 2,
+        });
+
+        if (signingKeyOutput.length == 0) {
+          console.log(`
+            access-control: gate: content with playbackId ${playbackId} is gated but corresponding public key not found for key ${req.body.pub}, disallowing playback
+          `);
+          throw new ForbiddenError(
+            "Content is gated and corresponding public key not found"
+          );
+        }
+
+        if (signingKeyOutput.length > 1) {
+          let collisionKeys = JSON.stringify(signingKeyOutput);
+          console.log(`
+            access-control: gate: content contentId ${content.id} with playbackId=${playbackId} is gated but multiple (${signingKeyOutput.length}) public keys found for key ${req.body.pub}, disallowing playback, colliding keys=${collisionKeys}
+          `);
+          throw new BadRequestError(
+            "Multiple signing keys found for the same public key."
+          );
+        }
+
+        const signingKey = signingKeyOutput[0];
+
+        if (signingKey.userId !== content.userId) {
+          console.log(`
+            access-control: gate: disallowing playback for contentId=${content.id} with playbackId=${playbackId} the content and the public key pub=${req.body.pub} do not share the same owner, signingKeyUserId=${signingKey.userId}, contentUserId=${content.userId}
+          `);
+          throw new NotFoundError("Content not found");
+        }
+
+        if (signingKey.disabled || signingKey.deleted) {
+          const signingKeyLog = JSON.stringify(JSON.stringify(signingKey));
+          console.log(`
+            access-control: gate: disallowing playback for contentId=${content.id} with playbackId=${playbackId} the public key pub=${signingKey.id} is disabled or deleted, signingKey=${signingKeyLog}
+          `);
+          throw new ForbiddenError("The public key is disabled or deleted");
+        }
+
+        tracking.recordSigningKeyValidation(signingKey.id);
+        res.set("Cache-Control", "max-age=120,stale-while-revalidate=600");
+        res.status(204);
+        return res.end();
+      case "webhook":
+        if (!req.body.accessKey) {
+          console.log(`
+            access-control: gate: no accessKey provided for playbackId ${playbackId}, disallowing playback
+          `);
+          throw new ForbiddenError(
+            "Content is gated and requires an access key"
+          );
+        }
+        const webhook = await db.webhook.get(content.playbackPolicy.webhookId);
+        if (!webhook) {
```

Comment:
Also! We must check if the webhook can listen to the event `playback.accessControl` (or whatever). I know you haven't created it yet, but commenting here as a reminder.

Reply:
This is still needed 👀 

And thinking about it, maybe we should allow fetching an account webhook of the `playback.accessControl` type, WDYT? So it defaults to idk the oldest active webhook for that event type, if you don't specify it in the `playbackPolicy` 🤔